answerking_app/mixins/OrderItemMixins.py

```diff
@@ -0,0 +1,137 @@
+                        "quantity": quantity,
+                        "sub_total": item.price * quantity,
+                    },
+                )
+            except DataError:
+                return None
+        else:
+            try:
+                order.orderline_set.filter(item=item.id).update(
+                    quantity=quantity, sub_total=quantity * item.price
+                )
+            except DataError:
+                return None
+
+        serializer = self.get_serializer(
+            order, data=request.data, partial=partial
+        )
+        serializer.is_valid(raise_exception=True)
+
+        if not serializer:
+            error_msg: ErrorMessage = {
+                "error": {
+                    "message": "Resource update failure",
+                    "details": "Failed to add item to order",
+                }
+            }
+            return Response(
+                error_msg,
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        serializer.save()
+        response: ReturnDict = serializer.data
+
+        return Response(response, status=status.HTTP_200_OK)
+
+
+class OrderItemRemoveMixin:
+    def remove(self, request, *args, **kwargs) -> Response:
+        order_id: str = kwargs.get("order_id")
+        item_id: str = kwargs.get("item_id")
+
+        order: Order | None = Order.objects.get(pk=order_id)
+        item: Item | None = Item.objects.get(pk=item_id)
+
+        if not order or not item:
+            error_msg: ErrorMessage = {
+                "error": {
+                    "message": "Request failed",
+                    "details": "Object not found",
+                }
+            }
+            return Response(
+                error_msg,
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        if item not in order.order_items.all():
+            error_msg: ErrorMessage = {
+                "error": {
+                    "message": "Resource update failure",
+                    "details": "Item not in order",
+                }
+            }
+            return Response(
+                error_msg,
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        updated_order: Order | None = self.remove_item(order, item)
+        if not updated_order:
+            error_msg: ErrorMessage = {
+                "error": {
+                    "message": "Resource update failure",
+                    "details": "Failed to delete item from order",
+                }
+            }
+            return Response(
+                error_msg,
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        response: ReturnDict = OrderSerializer(updated_order).data
+
+        return Response(response, status=status.HTTP_200_OK)
+
+    def remove_item(self, order: Order, item: Item) -> Order:
+        order.order_items.remove(item)
+        order.calculate_total()
+        return order
```

Comment:
can this ever return None?

Reply:
no?

Reply:
then updated_order should not be tested for being None to raise error in line 118